tooling/log-redaction/main.go

```diff
@@ -0,0 +1,209 @@
+	}
+
+	opts := &Options{
+		Stamp: 1,
+	}
+
+	cmd.Flags().StringVar(&opts.ServiceConfigFile, "service-config-file", "config/config.yaml", "Path to the service configuration file.")
+	cmd.Flags().StringVar(&opts.RedactionConfigFile, "redaction-config-file", "config/config.redaction.yaml", "Path to the redaction configuration file.")
+	cmd.Flags().StringVar(&opts.Cloud, "cloud", "dev", "The name of the cloud.")
+	cmd.Flags().StringVar(&opts.Environment, "environment", "dev", "The name of the environment.")
+	cmd.Flags().StringVar(&opts.Region, "region", "eastus2", "The name of the region.")
+	cmd.Flags().IntVar(&opts.Stamp, "stamp", opts.Stamp, "Stamp value to use.")
+	cmd.Flags().StringVar(&opts.Ev2Cloud, "ev2-cloud", "public", "Cloud to use for Ev2 configuration.")
+	cmd.Flags().StringVar(&opts.RegionShortSuffix, "region-short-suffix", "", "Suffix to use for region short-name.")
+	cmd.Flags().BoolVar(&opts.Debug, "debug", false, "Enable debug output to see the resolved config structure.")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		return runLogRedaction(cmd.Context(), opts)
+	}
+
+	if err := cmd.Execute(); err != nil {
+		logrus.Error("command failed: %w", err)
+		os.Exit(1)
+	}
+}
+
+func runLogRedaction(ctx context.Context, opts *Options) error {
+	redactionConfig, err := loadRedactionConfig(opts.RedactionConfigFile)
+	if err != nil {
+		return fmt.Errorf("failed to load redaction config: %w", err)
+	}
+
+	c, err := config.NewConfigProvider(opts.ServiceConfigFile)
+	if err != nil {
+		return fmt.Errorf("failed to load config file: %w", err)
+	}
+
+	ev2Config, err := ev2config.ResolveConfig(opts.Ev2Cloud, opts.Region)
+	if err != nil {
+		return fmt.Errorf("failed to resolve ev2 config: %w", err)
+	}
+
+	replacements := &config.ConfigReplacements{
+		RegionReplacement:      opts.Region,
+		CloudReplacement:       opts.Cloud,
+		EnvironmentReplacement: opts.Environment,
+		StampReplacement:       strconv.Itoa(opts.Stamp),
+		RegionShortReplacement: opts.Region,
+		Ev2Config:              ev2Config,
+	}
+
+	resolver, err := c.GetResolver(replacements)
+
+	if err != nil {
+		return fmt.Errorf("failed to get resolver, %w", err)
+	}
+
+	cfg, err := resolver.GetConfiguration()
+	if err != nil {
+		return fmt.Errorf("Error resolving config, %w", err)
+	}
+
+	redactionKeys, err := resolveRedactionKeys(cfg, redactionConfig)
+	if err != nil {
+		return fmt.Errorf("failed to resolve redaction keys: %w", err)
+	}
+
+	redactionConfigTemplate, err := template.New("redactionConfig").Parse(string(mustGatherTemplate))
+	if err != nil {
+		return fmt.Errorf("failed to parse redaction config template: %w", err)
+	}
+
+	redactionConfigTemplate.Execute(os.Stdout, map[string]interface{}{
+		"redactionKeys": redactionKeys,
+	})
+
+	return nil
+}
+
+func resolveRedactionKeys(cfg config.Configuration, redactionConfig redactionConfig) ([]string, error) {
+	redactionKeys := make([]string, 0)
+	for _, redactionKey := range redactionConfig["keys"] {
```

Comment:
Clever, but a flat list with some repetition for keys with common prefixes would be easier to read, only mildly harder to write, and much easier to reason about as opposed to this :)

Reply:
After sleeping over it, agreed. I'll try to find a simpler approach